### PR TITLE
Connection to multiple servers

### PR DESCRIPTION
**Pull request by matteosz**
This PR aims to improve the reliability of the system by allowing the client to connect to multiple servers at the same time. In particular, this can be useful in the future to develop mechanisms to bypass censorship.

MultiConnection
----------------
Currently it implements a MultiConnection class that simply subscribes to the channel with each peer. In the future optimisations can be made to reduce traffic (e.g. exchanging ids with peers instead of whole messages).
The peers considered are only the ones to the "main" server, so successive GreetLao messages have no effect.

I tested it by connecting two times to the same server (Go) and to two local servers (Scala and Go).

### CODE REVIEW

**Review by simone-kalbermatter — DISMISSED** [review]
Impressive work! Clean, well documented and keeps the existing structure.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/MultiConnection.java
Comment (Warhedgehog) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/MultiConnection.java:1]: I'm missing the point of having the architecture done so. Why treating the first connection different than the others ? By that I mean that I would imagine the result to be simpler with simply populating the connection with every connection including the first one

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Impressive work and well documented 👍 A couple question on my end

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/LaoActivity.java [hunk 1]
```diff
@@ -277,15 +273,6 @@ private void setEventsTab() {
...
 
-  /** Restore the fragment contained in the stack as container of the current lao */
-  public void resetLastFragment() {
-    Fragment fragment = fragmentStack.pop();
-    getSupportFragmentManager()
-        .beginTransaction()
-        .replace(R.id.fragment_container_lao, fragment)
-        .commit();
-  }
-
```
[the remark this answers is not in the document]
Comment (matteosz) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/LaoActivity.java:288]: Yeah I just realised that I've committed some conflicting and unwanted changes due to a previous merge I've done 😅 , I'll fix that

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/MultiConnection.java
Comment (Warhedgehog) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/MultiConnection.java:1]: Makes sense 👍

**Review by Warhedgehog — APPROVED** [review]
LGTM, great work !

**Pull request merged**